torch/quantization/quantize_fx.py

```diff
@@ -0,0 +1,149 @@
+from .fx import Fuser  # noqa: F401
+from .fx import Quantizer  # noqa: F401
+
+def fuse_fx(graph_module, inplace=False):
+    fuser = Fuser()
+    return fuser.fuse(graph_module, inplace)
+
+def _prepare_fx(graph_module, qconfig_dict, inplace, is_dynamic):
+    quantizer = Quantizer()
+    prepare = quantizer.prepare_dynamic if is_dynamic else quantizer.prepare
+    prepared = prepare(graph_module, qconfig_dict, inplace)
+    return prepared
+
+def prepare_fx(graph_module, qconfig_dict, inplace=False):
+    """ If graph_module is in training mode, the model will be
+    prepared as a qat model, otherwise, it will be prepared as a
+    model used for post training static quantization.
+
+    Args:
+      graph_module: model from symbolic_tracing (torch.fx)
+      qconfig_dict: tbd
+    """
+    return _prepare_fx(graph_module, qconfig_dict, inplace, False)
+
+def prepare_dynamic_fx(graph_module, qconfig_dict, inplace=False):
+    return _prepare_fx(graph_module, qconfig_dict, inplace, True)
+
+def _convert_fx(graph_module, inplace=False, debug=False, is_dynamic=False):
+    quantizer = Quantizer()
+    return quantizer.convert(graph_module, inplace, debug, is_dynamic)
+
+def convert_fx(graph_module, inplace=False, debug=False):
+    return _convert_fx(graph_module, inplace, debug, is_dynamic=False)
+
+def convert_dynamic_fx(graph_module, inplace=False, debug=False):
+    return _convert_fx(graph_module, inplace, debug, is_dynamic=True)
+
+def _quantize_fx(model, qconfig_dict, run_fn=None, run_args=None, inplace=False, debug=False, is_dynamic=False):
+    if is_dynamic:
+        model = prepare_dynamic_fx(model, qconfig_dict, inplace)
+        # TODO: change inplace to True since the model is already copied in
+        # prepare
+        model = convert_dynamic_fx(model, False, debug)
+    else:
+        assert run_fn, "Must provide calibration function for post training static quantization"
+        assert run_args, "Must provide calibration dataset for post training static quantization"
+        model = prepare_fx(model, qconfig_dict, inplace)
+        run_fn(model, *run_args)
+        # TODO: change inplace to True since the model is already copied in
+        # prepare
+        model = convert_fx(model, False, debug)
+
+    return model
+
+
+def quantize_fx(model, qconfig_dict, run_fn, run_args, inplace=False, debug=False):
+    r"""Quantize the input float symbolically traced GraphModule model with
```

Comment:
> post training static quantization (if the model is in eval mode) or quantization aware training (if the model is in training mode).

This would be easy to mess up or forget, thoughts about either having separate functions for PTQ versus QAT, or forcing users to choose PTQ vs QAT in an argument and enforce that the `model.training` flag is set correctly based on the quantization mode?

Reply:
maybe we can also use this api only for post training static. i think for qat, most of the time people will be using prepare and convert

Reply:
sure, sounds reasonable